### PR TITLE
[AMD][aiter] Fix cuda_graph_kv_indices OOB under page_size>1

### PR DESCRIPTION
**Pull request by xiaobochen-amd**
## Motivation

The aiter attention backend crashes on ROCm with
`HSA_STATUS_ERROR_MEMORY_APERTURE_VIOLATION` /
`HIP error: an illegal memory access was encountered` once the running
batch's `seq_lens_sum` grows past the cuda-graph KV-indices buffer.
The crash typically lands at a downstream `torch.cat` /
`synchronize()` so it looks unrelated, but the actual fault is an OOB
`tl.store` inside `create_flashinfer_kv_indices_triton`.

Trigger conditions (must all hold):
- `--attention-backend aiter`, non-MLA model, `--page-size > 1`
- CUDA graph enabled (default)
- `seq_lens_sum` at decode time exceeds
  `cuda_graph_max_bs * ceil(max_context_len / page_size)` — easy to
  hit at moderate concurrency on long-context models, especially when
  `cuda_graph_max_bs` is small (e.g. clamped by
  `--max-running-requests`).

Root cause: `init_cuda_graph_state` allocates `cuda_graph_kv_indices`
at page granularity (`max_bs * ceil(max_context_len / page_size)`),
but every forward path that fills it (default decode, MLA decode,
target_verify, draft_extend{,_v2}) writes per token via
`create_flashinfer_kv_indices_triton` with
`kv_indptr = cumsum(seq_lens)`. With `--page-size > 1` the buffer is
under-sized by `page_size`x and the kernel writes past its end.

Workaround for affected versions (no code change): pass `--page-size 1`.

## Modifications

- `python/sglang/srt/layers/attention/aiter_backend.py`,
  `init_cuda_graph_state`: allocate
  `max_bs * max_num_blocks_per_seq * page_size` instead of
  `max_bs * max_num_blocks_per_seq`.
  - `>= max_bs * max_context_len`, the worst-case per-token write.
  - Still divisible by `max_num_blocks_per_seq`, so the unified
    attention path's `view(-1, max_num_blocks_per_seq)` stays valid
    even when `max_context_len % page_size != 0`.
  - Identical to old behavior at `page_size = 1`.
- Inline TODO captures the proper page-mode root fix
  (`paged_attention_ragged` still runs with `view(-1, 1, ...)` and
  `block_size=1`, so `--page-size > 1` does not actually engage the
  attention kernel today; making it engage requires a per-page indices
  kernel + all metadata sites + the `paged_attention_ragged` call site
  + FP8 KV coordination, after which this allocation can revert).
- No other call sites or kernels are touched.

## Accuracy Tests

Reproducer (deterministic crash without this PR):

```bash
python -m sglang.launch_server \
  --model-path MiniMax-M2.7 \
  --tp-size 2 \
  --attention-backend aiter \
  --page-size 64 \
  --max-running-requests 64 \
  --kv-cache-dtype fp8_e4m3





















<!-- pr-states:start -->
---
### CI States

Latest PR Test (Base): <!-- slot:pr-test:start -->:x: [Run #26309887455](https://github.com/sgl-project/sglang/actions/runs/26309887455)<!-- slot:pr-test:end -->
Latest PR Test (Extra): <!-- slot:pr-test-extra:start -->:x: [Run #26309887338](https://github.com/sgl-project/sglang/actions/runs/26309887338)<!-- slot:pr-test-extra:end -->
<!-- pr-states:end -->

### CODE REVIEW

File: python/sglang/srt/layers/attention/aiter_backend.py
Comment (kkHuang-amd) [inline python/sglang/srt/layers/attention/aiter_backend.py:2922]: This line also need to be modified.
max_kv_len = page_table.shape[1] * self.page_size => max_kv_len = page_table.shape[1]

Comment (kkHuang-amd) [inline python/sglang/srt/layers/attention/aiter_backend.py:2675]: Need to be modified

max_kv_len = page_table.shape[1] * self.page_size => max_kv_len = page_table.shape[1]

**Comment by fanxingran** [comment]
@kkHuang-amd 
The issue has been fixed. Could you please review it again? Thanks.

**Review by kkHuang-amd — APPROVED** [review]
LGTM

**Review by HaiShaw — APPROVED** [review]

**Comment by HaiShaw** [comment]
@amd-bot ci-status

**Comment by HaiShaw** [comment]
@amd-bot ci-status
